src/panels/lovelace/editor/card-editor/hui-card-picker.ts

```diff
@@ -95,17 +318,107 @@ export class HuiCardPicker extends LitElement {
   }
 
   private async _cardPicked(ev: Event): Promise<void> {
-    const type = (ev.currentTarget! as CardPickTarget).type;
+    const config: LovelaceCardConfig = (ev.currentTarget! as CardPickTarget)
+      .config;
 
-    const elClass = await getCardElementClass(type);
-    let config: LovelaceCardConfig = { type };
+    fireEvent(this, "config-changed", { config });
+  }
+
+  private async _getCardConfig(
+    cardConfig: CardPickerConfig
+  ): Promise<LovelaceCardConfig | undefined> {
+    if (!this.hass) {
+      return undefined;
+    }
+
+    const elClass = await getCardElementClass(
+      cardConfig.lovelaceCardConfig.type
+    );
+    let config: LovelaceCardConfig = cardConfig.lovelaceCardConfig;
 
     if (elClass && elClass.getStubConfig) {
-      const cardConfig = elClass.getStubConfig(this.hass!);
-      config = { ...config, ...cardConfig };
+      const stubConfig = elClass.getStubConfig(this.hass!);
```

Comment:
I think using this, I can probably remove the Lovelace Card object in the cardConfigs Array. Just have `type: ""`

This should pull all the defaults except for entities.